src/EditTableDialog.cpp

```diff
@@ -329,6 +337,9 @@ void EditTableDialog::populateConstraints()
         name->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled | Qt::ItemIsEditable);
         ui->tableConstraints->setItem(row, kConstraintName, name);
 
+        // Reference
+        ui->tableConstraints->setItemDelegateForColumn(kConstraintReference, m_fkEditorDelegate);
```

Comment:
This should be moved to the constructor because you only need to apply the delegate once.

I think you also need to create a new foreign key editor delegate object for the constraints view, i.e. you can't reuse ```m_fkEditorDelegate```. Instead you must create a new object (maybe ```m_fkEditorConstraintsDelegate``` or so) and use that for the constraints view. The [Qt documentation](https://doc.qt.io/qt-5/qabstractitemview.html#setItemDelegateForColumn) has a paragraph about this:
> Warning: You should not share the same instance of a delegate between views. Doing so can cause incorrect or unintuitive editing behavior since each view connected to a given delegate may receive the closeEditor() signal, and attempt to access, modify or close an editor that has already been closed.

I haven't tried it but this might fix the problem that the text doesn't show up initially and only becomes visible after double clicking the item. Not sure though.